### PR TITLE
✨ Add RLP decode result

### PR DESCRIPTION
**Pull request by zander-xyz**

### CODE REVIEW

File: include/monad/rlp/decode_error.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+MONAD_RLP_NAMESPACE_END
+
+BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE_BEGIN
+template <>
+struct quick_status_code_from_enum<monad::rlp::DecodeError>
+    : quick_status_code_from_enum_defaults<monad::rlp::DecodeError>
+{
+    static constexpr auto const domain_name = "Decode Error";
+    static constexpr auto const domain_uuid =
+        "da6c5e8cd6a1101c9cff97a0f36ddcb9";
```
Comment (rgarc) [inline include/monad/rlp/decode_error.hpp]: where is this uuid from?

File: include/monad/rlp/decode_error.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#include <monad/core/result.hpp>
+#include <monad/rlp/config.hpp>
+
+#include <boost/outcome/experimental/status-code/quick_status_code_from_enum.hpp>
+
+MONAD_RLP_NAMESPACE_BEGIN
+
+enum class DecodeError
+{
+    SUCCESS = 0
```
Comment (rgarc) [inline include/monad/rlp/decode_error.hpp]: Why is `SUCCESS` a `DecodeError`?

Comment (zander-xyz) [reply include/monad/rlp/decode_error.hpp]: 0 needs to be success decode error per spec

File: include/monad/rlp/decode_error.hpp (hunk 1 above)
Comment (ned14) [reply include/monad/rlp/decode_error.hpp]: It should come from `random.org`

Comment (zander-xyz) [reply include/monad/rlp/decode_error.hpp]: random.org

**Pull request merged**